lib/booking_repository.py

```diff
@@ -50,6 +52,30 @@ def show_property_bookings(self,property_id):
     def change_booking_approval(self, id):
         self._connection.execute("UPDATE bookings SET approved = 'TRUE' WHERE id = %s" , (id,))
         return None
+    
+    def dates_taken(self, property_id):
+        bookings = self.show_property_bookings(property_id)
+        dates_taken = []
+        for booking in bookings:
+            if booking.approved:
+                dates_taken += self._dates_in_range(booking.dates_booked_from, booking.dates_booked_to)
+        return dates_taken
+
+    def _dates_in_range(self, date_from, date_to=None):
+        if not date_to:
+            date_from = datetime.strptime(date_from, "%Y-%m-%d")
+            return [date_from.strftime("%d/%m/%Y")]
+
+        date_from = datetime.strptime(date_from, "%Y-%m-%d")
+        date_to = datetime.strptime(date_to, "%Y-%m-%d")
+        
+        timestamps = pd.date_range(date_from, date_to).tolist()
```

Comment:
Is this timestamp in seconds? or is the output still date?

Reply:
I think seconds will be there but it will be 00:00:00. I think it needs to be in that format for the applied methods to work. Don't need to call the variable "timestamps" necessarily. I think it was more because I was trying to get my head around how it worked. In practical terms, they are dates.